coordinator/src/trade/mod.rs

```diff
@@ -173,6 +207,66 @@ impl TradeExecutor {
         };
     }
 
+    /// Settles the accepted invoice for the given trader
+    async fn settle_invoice(&self, trader: PublicKey, order_id: Uuid) -> Result<()> {
+        let pre_image = spawn_blocking({
+            let pool = self.node.pool.clone();
+            move || {
+                let mut conn = pool.get()?;
+                let pre_image =
+                    db::hodl_invoice::update_hodl_invoice_to_settled(&mut conn, order_id)?;
+
+                anyhow::Ok(pre_image)
+            }
+        })
+        .await??
+        .context("Missing pre_image")?;
+
+        let pre_image = PreImage::from_url_safe_encoded_pre_image(&pre_image)?;
+
+        self.node
+            .lnd_bridge
+            .settle_invoice(pre_image.get_base64_encoded_pre_image())
+            .await?;
```

Comment:
🔧 I guess we would like to update the DB only if we were able to settle the invoice with the LND bridge. Could we use a transaction here?

Reply:
It shouldn't matter as the error handling will set the hodl_invoice to `Failed`.